cxx-qt/src/lib.rs

```diff
@@ -60,6 +64,38 @@ pub fn signals(_args: TokenStream, _input: TokenStream) -> TokenStream {
     unreachable!("cxx_qt::signals should not be used as a macro by itself. Instead it should be used within a cxx_qt::bridge definition")
 }
 
+/// A macro which describes that the inner methods should be implemented on the C++ QObject.
+/// This allows for defining C++ methods and Q_INVOKABLE methods for QML in Rust.
+///
+/// It should not be used by itself and instead should be used inside a cxx_qt::bridge definition.
+///
+/// # Example
+///
+/// ```ignore
+/// #[cxx_qt::bridge]
+/// mod my_object {
+///     #[derive(Default)]
+///     struct Data {
+///         property: i32,
+///     }
+///
+///     #[derive(Default)]
+///     struct RustObj;
+///
+///     impl cxx_qt::QObject<RustObj> {
+///         #[invokable]
+///         fn invokable(&self, a: i32, b: i32) -> i32 {
+///             a + b
+///         }
+///     }
+/// }
+/// ```
+#[proc_macro]
+#[allow(non_snake_case)]
+pub fn QObject(_input: TokenStream) -> TokenStream {
```

Comment:
This shouldn't be a macro, it needs to be a type:
``` rust
pub struct QObject<T> {
    phantom: std::marker::PhantomData<T>
}
```

Reply:
We can't export non macro things in a proc-macro crate, so it has to be this.